### ISSUE TITLE
[Bug]: SWC Picker autofocus does not show focus ring when navigating via mouse

### ISSUE THREAD

**Issue by datle-Adobe** [adobe/spectrum-web-components#4671]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

SWC Picker

### Expected behavior

When navigating to Picker via keyboard and `autofocus` property is set to true, I should see the focus ring. This is consistent with SWC Textfield when the same property is set

### Actual behavior

When navigating to Picker via keyboard and `autofocus` property is set to true, I don't see the focus ring

### Screenshots

Using this link: https://studio.webcomponents.dev/edit/q4K3CJcbrJcFIr8gildQ/src/index.ts?p=stories
![Screen Recording 2024-08-14 at 1 09 30 PM](https://github.com/user-attachments/assets/43c95378-e4f0-4290-a3ec-623749f55a59)

### What browsers are you seeing the problem in?

Chrome, Safari

### How can we reproduce this issue?

1. Go to https://studio.webcomponents.dev/edit/q4K3CJcbrJcFIr8gildQ/src/index.ts?p=stories
2. Click on 'Dialog with autofocus select' button
3. See the error where the focus ring is not visible even though it's focused

### Sample code or abstract reproduction which illustrates the problem

The same code below can be found in https://studio.webcomponents.dev/edit/q4K3CJcbrJcFIr8gildQ/src/index.ts?p=stories
```
<overlay-trigger type="modal">
          <sp-dialog-wrapper
              slot="click-content"
              underlay
              headline="Dialog title"
              mode="fullscreen"
              confirm-label="Keep Both"
              secondary-label="Replace"
              cancel-label="Cancel"
              footer="Content for footer"
          >
            <sp-field-label for="picker-m" size="m">Selection type:</sp-field-label>
            <sp-picker id="picker-1" size="m" label="Selection type" autofocus>
              <sp-menu-item>Deselect</sp-menu-item>
              <sp-menu-item>Select inverse</sp-menu-item>
              <sp-menu-item>Feather...</sp-menu-item>
              <sp-menu-item>Select and mask...</sp-menu-item>
              <sp-menu-divider></sp-menu-divider>
              <sp-menu-item>Save selection</sp-menu-item>
              <sp-menu-item disabled>Make work path</sp-menu-item>
            </sp-picker>
          </sp-dialog-wrapper>
          <sp-button
              slot="trigger"
          >
              Dialog with autofocus select
          </sp-button>
        </overlay-trigger>
```

### Severity

SEV 3

**Comment by najikahalsema** [comment]
We don't support an autofocus API on our Picker, which would explain why this wouldn't work.
